crates/csharp/src/csharp_ident.rs

```diff
@@ -0,0 +1,110 @@
+use heck::{ToLowerCamelCase, ToUpperCamelCase};
+
+pub(crate) trait ToCSharpIdent: ToOwned {
+    fn csharp_keywords() -> Vec<&'static str>;
+    fn to_csharp_ident(&self) -> Self::Owned;
+    fn to_csharp_ident_upper(&self) -> Self::Owned;
+}
+
+impl ToCSharpIdent for str {
+    // Source: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/
+    fn csharp_keywords() -> Vec<&'static str> {
```

Comment:
It would be more efficient to make this return a `&'static [&'static str]` instead of allocating and populating a fresh new new `Vec` every time it's called.  Something like:
```rust
fn csharp_keywords() -> &'static [&'static str] {
    static ARRAY: &[&str] = &["abstract", ...];
    ARRAY
}
```

Reply:
Since that requires a bit more refactoring than just moving. I'll follow up on it with another PR